manifests/mongodb.pp

```diff
@@ -0,0 +1,105 @@
+# Certs configurations for MongoDB
+class certs::mongodb (
+  $hostname             = $certs::node_fqdn,
+  $cname                = $certs::cname,
+  $generate             = $certs::generate,
+  $regenerate           = $certs::regenerate,
+  $deploy               = $certs::deploy,
+  $pki_dir              = $certs::pki_dir,
+  $server_cert          = $certs::server_cert,
+  $country              = $certs::country,
+  $state                = $certs::state,
+  $city                 = $certs::city,
+  $org                  = $certs::org,
+  $org_unit             = $certs::org_unit,
+  $expiration           = $certs::expiration,
+  $default_ca           = $certs::default_ca,
+  $ca_key_password_file = $certs::ca_key_password_file,
+  $group                = $certs::group,
+) inherits certs {
+
+  $mongodb_server_cert_name = 'mongodb-server-certificate'
+  $mongodb_server_bundle = "${pki_dir}/mongodb/${mongodb_server_cert_name}-bundle.pem"
+  $mongodb_server_ca_cert = $certs::katello_server_ca_cert
+
+  $mongodb_client_cert_name = 'mongodb-client-certificate'
+  $mongodb_client_cert = "/etc/pulp/mongodb/${mongodb_client_cert_name}.crt"
+  $mongodb_client_key  = "/etc/pulp/mongodb/${mongodb_client_cert_name}.key"
+  $mongodb_client_ca_cert = $certs::katello_server_ca_cert
+
+  cert { $mongodb_server_cert_name:
+    ensure        => present,
+    hostname      => $hostname,
+    cname         => $cname,
+    country       => $country,
+    state         => $state,
+    city          => $city,
+    org           => $org,
+    org_unit      => $org_unit,
+    expiration    => $expiration,
+    ca            => $default_ca,
+    generate      => $generate,
+    regenerate    => $regenerate,
+    deploy        => $deploy,
+    password_file => $ca_key_password_file,
+  }
+
+  cert { $mongodb_client_cert_name:
+    ensure        => present,
+    hostname      => $hostname,
+    cname         => $cname,
+    purpose       => 'client',
+    country       => $country,
+    state         => $state,
+    city          => $city,
+    org           => $org,
+    org_unit      => $org_unit,
+    expiration    => $expiration,
+    ca            => $default_ca,
+    generate      => $generate,
+    regenerate    => $regenerate,
+    deploy        => $deploy,
+    password_file => $ca_key_password_file,
+  }
+
+  if $deploy {
+    file { "${pki_dir}/mongodb":
+      ensure => directory,
+      mode   => '0750',
+      owner  => 'mongodb',
+      group  => 'mongodb',
+    }
+
+    file { '/etc/pulp/mongodb':
```

Comment:
Just to show the relation, I'd store this in a variable (`$mongodb_client_cert_dir`) and use it in `$mongodb_client_{cert,key}` as well. It doesn't have to be a class parameter, but I think it might even make sense.

Reply:
I might still make that change, I can't get these tests to run locally so some of the code pushes you see are just to use Travis for acceptance testing.